lib/ui/painting.dart

```diff
@@ -2869,44 +2910,68 @@ class ImageFilter {
 
   final Float64List _data;
   final FilterQuality? _filterQuality;
+  final ColorFilter? _colorFilter;
+  final ImageFilter? _innerFilter;
+  final ImageFilter? _outerFilter;
   final int _type;
-  _ImageFilter? _nativeFilter;
+  late final _ImageFilter _nativeFilter = _makeNativeImageFilter();
 
   // The type of SkImageFilter class to create for Skia.
-  static const int _kTypeBlur = 0;   // MakeBlurFilter
-  static const int _kTypeMatrix = 1; // MakeMatrixFilterRowMajor255
+  static const int _kTypeComposed = 0;    //SkImageFilters::Compose
+  static const int _kTypeBlur = 1;        // MakeBlurFilter
+  static const int _kTypeMatrix = 2;      // MakeMatrixFilterRowMajor255
+  static const int _kTypeColorFilter = 3; //SkImageFilters::ColorFilter
+
+  /// Creates a new [ImageFilter] from combining the effects of this filter and
+  /// `outer`.
+  ///
+  /// This method returns a new filter that has the same effect as applying this
+  /// filter and the `outer` filter subsequently, i.e.,
+  /// result = outer(this(source)).
+  ImageFilter compose(ImageFilter outer) => ImageFilter.compose(outer, this);
```

Comment:
I don't like the name `compose` primarily because it doesn't indicate which direction the composition happens in.

In particular, the Skia compose method on an `SkImageFilter` takes an `inner` argument, so right here we have examples of two engineers having opposing default concepts of which direction a `compose()` method would do the composition.

While code completion in IDEs might help with this by exposing the name of the parameter, I'm finding that many IDEs can be finicky in whether or not they show the parameter list and so that is unreliable.

Changing to a named parameter might help with this, but perhaps a pair of methods that allow the developer to compose the chains in either direction and with direction-specific names might be more developer-friendly?

Reply:
Removed the instance method version because of analyzer complaints. The composing direction should be unopinionated in the constructor now.

Reply:
What were the analyzer complaints?

Reply:
FWIW, we call something like this `combine` on `Path`.

Reply:
Resolved with the named parameters on the new constructor.